include/boost/geometry/algorithms/detail/is_valid/polygon.hpp

```diff
@@ -111,14 +111,10 @@ class is_valid_polygon
                                          VisitPolicy& visitor,
                                          Strategy const& strategy)
     {
-        return
-            detail::check_iterator_range
-                <
-                    per_ring<VisitPolicy, Strategy>,
-                    true // allow for empty interior ring range
-                >::apply(boost::begin(interior_rings),
-                         boost::end(interior_rings),
-                         per_ring<VisitPolicy, Strategy>(visitor, strategy));
+        return boost::end(interior_rings) == std::find_if(
+            boost::begin(interior_rings),
+            boost::end(interior_rings),
+            per_ring<VisitPolicy, Strategy>(visitor, strategy));
```

Comment:
The behavior is the same. I'd only think about creating `end` once and replacing `per_ring` to `is_invalid_ring`.